src/apps/KintoApp.sol

```diff
@@ -0,0 +1,278 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721EnumerableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721BurnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/ECDSAUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import "../interfaces/IKintoID.sol";
+import "../interfaces/IKintoApp.sol";
+import "../interfaces/IKintoWalletFactory.sol";
+
+// import "forge-std/console2.sol";
+
+/**
+ * @title KintoApp
+ * @dev A contract that holds all the information of a KintoApp
+ */
+contract KintoApp is
+    Initializable,
+    ERC721Upgradeable,
+    ERC721EnumerableUpgradeable,
+    ERC721BurnableUpgradeable,
+    AccessControlUpgradeable,
+    UUPSUpgradeable,
+    IKintoApp
+{
+    /* ============ Constants ============ */
+    bytes32 public constant override UPGRADER_ROLE = keccak256("UPGRADER_ROLE");
+    bytes32 public constant override DEVELOPER_ADMIN = keccak256("DEVELOPER_ADMIN");
+
+    uint256 public constant RATE_LIMIT_PERIOD = 1 minutes;
+    uint256 public constant RATE_LIMIT_THRESHOLD = 10;
+    uint256 public constant GAS_LIMIT_PERIOD = 30 days;
+    uint256 public constant GAS_LIMIT_THRESHOLD = 1e16; // 0.01 ETH
+
+    /* ============ State Variables ============ */
+
+    uint256 private _nextTokenId;
+
+    mapping(address => IKintoApp.Metadata) public appMetadata;
+    mapping(address => address) public childToParentContract;
+    mapping(address => mapping(address => bool)) public appSponsoredContracts; // other contracts to be sponsored
+
+    /* ============ Events ============ */
```

Comment:
```suggestion
    /* ============ Events ============ */
```

Reply:
Not sure what you meant here but the change is the same